[case testModel]
# flags: --strict
from typing import Optional

from aio_rom import Model

class User(Model):
    id: str
    name: str

    def __init__(self, id: str, name: str):
        self.id = id
        self.name = name

oi: Optional[int]

User("1")  # E: Missing positional argument "name" in call to "User"
User("1", "blah")
User("1", 2)  # E: Argument 2 to "User" has incompatible type "int"; expected "str"
User(id=str(), name=str())
User(id=oi, name="")  # E: Argument "id" to "User" has incompatible type "Optional[int]"; expected "str"
reveal_type(User.NotFoundException)  # N: Revealed type is "Type[aio_rom.exception.ModelNotFoundException]"
reveal_type(User.NotFoundException())  # N: Revealed type is "aio_rom.exception.ModelNotFoundException"
